mlrun/config.py

```diff
@@ -1237,9 +1237,16 @@ def get_s3_storage_options(self) -> dict[str, typing.Any]:
 
         return storage_options
 
-    def is_explicit_ack(self, version=None) -> bool:
+    def is_explicit_ack_enabled(self) -> bool:
+        version = None
+        if is_running_as_api():
+            import server.api.utils.runtimes.nuclio
+
+            version = server.api.utils.runtimes.nuclio.resolve_nuclio_version()
```

Comment:
importing server from the client is not allowed

Reply:
That's why it's inside `if is_running_as_api()`, to ensure it only runs in server. If that's not a good solution, can you briefly explain why and suggest an alternative?

Reply:
I see that it's enforced by the linter, so I guess I'll move `resolve_nuclio_version` to `mlrun`.

Reply:
I see that the linter enforces this. Not 100% sure how to work around it yet.

Reply:
Moved nuclio version resolution to server init, where it writes the nuclio version to the config, as is done with other configuration such as `config.httpdb.state`.